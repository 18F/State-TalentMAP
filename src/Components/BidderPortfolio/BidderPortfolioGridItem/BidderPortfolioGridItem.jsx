import React from 'react';
import { BIDDER_OBJECT } from '../../../Constants/PropTypes';
import CurrentUserGeneralInformation from '../../ProfileDashboard/CurrentUser/CurrentUserGeneralInformation';
import CurrentUserPersonalInformation from '../../ProfileDashboard/CurrentUser/CurrentUserPersonalInformation';
import CurrentUserBidInformation from '../../ProfileDashboard/CurrentUser/CurrentUserBidInformation';
import BidderPortfolioViewMore from '../BidderPortfolioViewMore';
import CheckBox from '../../CheckBox';

const BidderPortfolioGridItem = ({ userProfile }) => (
  <div className="usa-grid-full current-user bidder-portfolio-grid-item">
    <div className="usa-width-one-fourth grid-item-section">
      <div className="checkbox-container">
        <CheckBox id={`checkbox-${userProfile.id}`} label="Select this user" value={false} labelSrOnly small />
      </div>
      <div className="general-information-container" >
        <CurrentUserGeneralInformation
          userProfile={userProfile}
          showEditLink={false}
          useGroup
        />
      </div>
    </div>
    <div
      className={`usa-width-one-fourth grid-item-section
        current-user-personal-information-grid-container`}
    >
      <CurrentUserPersonalInformation userProfile={userProfile} />
    </div>
    <div className="usa-width-one-fourth grid-item-section">
      <CurrentUserBidInformation />
    </div>
    <div className="usa-width-one-fourth grid-item-section">
      <BidderPortfolioViewMore className="white-button" />
    </div>
  </div>
);

BidderPortfolioGridItem.propTypes = {
  userProfile: BIDDER_OBJECT.isRequired,
};

export default BidderPortfolioGridItem;
