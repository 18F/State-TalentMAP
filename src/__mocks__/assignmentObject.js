const assignmentObject = {
  start_date: '10-10-2016',
  estimated_end_date: '10-10-2018',
  position: 'Specialist',
};

export default assignmentObject;
