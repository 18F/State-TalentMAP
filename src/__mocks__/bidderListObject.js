const bidderListObject = {
  count: 3,
  results: [
    {
      id: 1,
      skill_code: 'INFORMATION MANAGEMENT (2880)',
      user: {
        first_name: 'John',
        last_name: 'Doe',
        email: 'doej@state.gov',
      },
      is_cdo: false,
      bid_statistics: [
        {
          draft: 1,
          submitted: 1,
        },
      ],
    },
    {
      id: 2,
      skill_code: 'INFORMATION MANAGEMENT (2880)',
      user: {
        first_name: 'Jane',
        last_name: 'Doe',
        email: 'doejane@state.gov',
      },
      is_cdo: false,
      bid_statistics: [
        {
          draft: 1,
          submitted: 1,
        },
      ],
    },
    {
      id: 3,
      skill_code: 'INFORMATION MANAGEMENT (2880)',
      user: {
        first_name: 'John',
        last_name: 'Smith',
        email: 'smithj@state.gov',
      },
      is_cdo: false,
      bid_statistics: [
        {
          draft: 1,
          submitted: 1,
        },
      ],
    },
  ],
};

export default bidderListObject;
