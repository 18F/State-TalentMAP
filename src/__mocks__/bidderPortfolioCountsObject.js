const bidderPortfolioCountsObject = {
  all: 100,
  bidding: 50,
  inpanel: 20,
  inpost: 30,
};

export default bidderPortfolioCountsObject;
