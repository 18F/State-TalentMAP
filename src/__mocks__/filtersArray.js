const filters = [
  { item: { description: 'language' }, data: [{ code: 'a', custom_description: 'letter A' }] },
  { item: { description: 'skill' }, data: [{ code: '1', custom_description: '1' }] },
  { item: { description: 'grade' }, data: [{ code: '2', custom_description: '2' }] },
  { item: { description: 'region' }, data: [{ code: 'bur', custom_description: 'bureau' }] }];

export default filters;
