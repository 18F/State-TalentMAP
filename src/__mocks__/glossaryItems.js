const glossaryItems = [
  { id: 2, title: 'Assignment Officer (AO)', definition: 'The employee responsible for filling the position.' },
  { id: 3, title: 'Bidder', definition: 'An employee who is applying for a position.' },
  { id: 4, title: 'Cable', definition: 'A secure text-based message sent from one State department location to another.' },
  { id: 5, title: 'Bureau', definition: 'The organization sponsoring a position. Most bureaus oversee regional areas, such as Western Europe. Others, such as the Bureau of Consular Affairs, oversee functional groups. If referring to a regional bureau, include the acronym -- State Department employees often use acronyms to refer to regional bureaus. In UI copy, put the acronym before the full name, eg: (WHA) Bureau of Western Hemispheric Affairs.' },
  { id: 6, title: 'Career development officer (CDO)', definition: 'An employee responsible for helping others find and successfully bid for appropriate positions for their career goals.' },
  { id: 7, title: 'Career Track', definition: 'A chosen professional competency.' },
  { id: 8, title: 'Cone', definition: 'A cone is a colloquial synonym for a generalist or specialist functional field. Bids can be "in cone" (for a position whose skill code belongs to the bidders\' specialization) or "out of cone." Each cone contains multiple skill codes.' },
  { id: 9, title: 'Cost of living allowance (COLA)', definition: 'A salary increase to reimburse employees for prices at a foreign post that are higher than same costs in Washington, DC. The specific value of the allowance changes based on local prices. COLA is a colloquial shorthand for what is officially known as post allowance.' },
];

export default glossaryItems;
