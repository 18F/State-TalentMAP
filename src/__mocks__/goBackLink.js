const goBackLink = { link: '/obc/post/100', text: 'Go back to Post details' };

export default goBackLink;
