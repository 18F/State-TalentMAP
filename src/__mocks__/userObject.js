const userObject = {
  skill_code: 'INFORMATION MANAGEMENT (2880)',
  user: {
    first_name: 'John',
    last_name: 'Doe',
    email: 'doej@state.gov',
  },
  is_cdo: true,
};

export default userObject;
