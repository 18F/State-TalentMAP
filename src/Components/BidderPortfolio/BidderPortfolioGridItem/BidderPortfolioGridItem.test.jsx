import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import BidderPortfolioGridItem from './BidderPortfolioGridItem';
import { bidderUserObject } from '../../../__mocks__/userObject';

describe('BidderPortfolioGridItemComponent', () => {
  it('is defined', () => {
    const wrapper = shallow(<BidderPortfolioGridItem userProfile={bidderUserObject} />);
    expect(wrapper).toBeDefined();
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<BidderPortfolioGridItem userProfile={bidderUserObject} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
