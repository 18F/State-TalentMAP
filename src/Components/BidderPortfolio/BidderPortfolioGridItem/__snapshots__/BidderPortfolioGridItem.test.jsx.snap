// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BidderPortfolioGridItemComponent matches snapshot 1`] = `
<div
  className="usa-grid-full current-user bidder-portfolio-grid-item"
>
  <div
    className="usa-width-one-fourth grid-item-section"
  >
    <div
      className="checkbox-container"
    >
      <CheckBox
        id="checkbox-undefined"
        label="Select this user"
        labelSrOnly={true}
        name=""
        onCheckBoxClick={[Function]}
        title=""
        value={false}
      />
    </div>
    <div
      className="general-information-container"
    >
      <CurrentUserGeneralInformation
        showEditLink={false}
        useGroup={true}
        userProfile={
          Object {
            "is_cdo": false,
            "skill_code": "INFORMATION MANAGEMENT (2880)",
            "user": Object {
              "email": "doej@state.gov",
              "first_name": "John",
              "last_name": "Doe",
            },
          }
        }
      />
    </div>
  </div>
  <div
    className="usa-width-one-fourth grid-item-section
          current-user-personal-information-grid-container"
  >
    <CurrentUserPersonalInformation
      userProfile={
        Object {
          "is_cdo": false,
          "skill_code": "INFORMATION MANAGEMENT (2880)",
          "user": Object {
            "email": "doej@state.gov",
            "first_name": "John",
            "last_name": "Doe",
          },
        }
      }
    />
  </div>
  <div
    className="usa-width-one-fourth grid-item-section"
  >
    <CurrentUserBidInformation />
  </div>
  <div
    className="usa-width-one-fourth grid-item-section"
  >
    <BidderPortfolioViewMore
      className="white-button"
      useLink={false}
    />
  </div>
</div>
`;
